### PR TITLE
refactor: drop poetry and use newer package manifest loader

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did

Was a little annoyed using poetry, this refactors back to using it with standard format tools (also works w/ `uv pip`)

### CODE REVIEW

**Comment by fubuloubu** [comment]
forgot to add the work I had refactoring to better ways of using package resources loader

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -52,12 +54,13 @@ jobs:
       - uses: actions/checkout@v4
       - uses: foundry-rs/foundry-toolchain@v1
       - uses: ApeWorX/github-action@v3
```
Comment (mikeshultz) [inline .github/workflows/test.yaml:56]: question: wait, what is this?

File: pyproject.toml [hunk 2]
```diff
@@ -1,34 +1,54 @@
...
-black = "^24.4.2"
+[tool.setuptools.package-data]
+apepay = [ "manifest.json", "py.typed" ]
 
-[build-system]
-requires = ["poetry-core"]
-build-backend = "poetry.core.masonry.api"
+[tool.setuptools_scm]
+# NOTE: Config entry needed for this plugin to function
+version_file = "sdk/py/apepay/version.py"
```
Comment (mikeshultz) [inline pyproject.toml:47]: suggestion: You should be able to get things running without this file. Unless this file is used within the SDK or something.

If it complains about not being able to find version, you might need to set the `root`. For example:

```toml
[tool.setuptools_scm]
root = "../.."
```

Assuming it thinks `sdk/py` is the root.

**Review by mikeshultz — APPROVED** [review]
Bit of commentary. Nothing blocking.  LGTM

File: .github/workflows/test.yaml (hunk 1 above)
Comment (fubuloubu) [reply .github/workflows/test.yaml:56]: we have a github action!
https://github.com/ApeWorX/github-action

File: pyproject.toml (hunk 2 above)
Comment (fubuloubu) [reply pyproject.toml:47]: ah okay, yeah this seems like a better way to configure. will do

Comment (fubuloubu) [reply pyproject.toml:47]: okay, yeah this isn't even needed

I was having issues with using `tool.setuptools-scm` as a key, which never works, and that was one attempt I made at resolving that

removing

File: .github/workflows/test.yaml (hunk 1 above)
Comment (mikeshultz) [reply .github/workflows/test.yaml:56]: This is cool

**Pull request merged**